webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -60,15 +66,16 @@
     file in smartling.
     */
     private static final int DEFAULT_BATCH_SIZE = 5004;
-    private static final List<String> EN_LOCALE = ImmutableList.of("en");
+    private static final String LOCALE_EN = "en";
 
     private final SmartlingClient smartlingClient;
     private final AssetPathAndTextUnitNameKeys assetPathAndTextUnitNameKeys;
     private final TextUnitSearcher textUnitSearcher; // TODO - Add a cached version of textunitsearcher
     private final TextUnitBatchImporterService textUnitBatchImporterService;
     private final Integer batchSize;
 
-    private List<SmartlingFile> lastPushResult;
+    private List<SmartlingFile> lastPushResult = new ArrayList<>();
+    private List<SmartlingFile> lastPushTranslationResult = new ArrayList<>();
```

Comment:
actually why do we maintain state in the spring component? how is this working with multiple concurrent calls?

Reply:
I was planning to have this resolved as part of the upcoming file upload PR. But I'll share my ideas here first to know your thoughts:

- One option is to remove the Component annotations from the `ThirdPartyTMS` implementors and create a new Component, say `ThirdPartyTMSFactory` to build instances of the `ThirdPartyTMS` implementors (through the factory) in the `ThirdPartyService` each time we call `syncMojitoWithThirdPartyTMS`, moving the conditionals we already have to this factory class (This would be my preferred approach)

- The second option is to add a `requestId` parameter to each of these methods and store the results based off that `requestId` in a Map. Obviously, the problem there is that we'll be storing a lot of data for single use so it's not a scalable solution.

Let me know what you think.

Reply:
if stated can't be moved easily into the functions you can use @Configurable instead of @Component + creating new instance with `new` (similar to option 1)